pkg/cli/clivalidation.go

```diff
@@ -101,10 +101,34 @@ func RequireEnvironmentName(cmd *cobra.Command, args []string, workspace workspa
 	return environmentName, err
 }
 
+// RequireEnvironmentNameOrID checks if an environment name or ID is provided as a flag or as a default environment in the workspace,
+// and returns an error if neither is present. It also handles any errors that occur while parsing the resource.
+func RequireEnvironmentNameOrID(cmd *cobra.Command, args []string, workspace workspaces.Workspace) (string, error) {
+	environmentNameOrID, err := cmd.Flags().GetString("environment")
+	if err != nil {
+		return "", err
+	}
+
+	// We store the environment id in config
+	if environmentNameOrID == "" && workspace.Environment != "" {
+		environmentNameOrID = workspace.Environment
+	}
+
+	if environmentNameOrID == "" && workspace.IsEditableWorkspace() {
+		// Setting a default environment only applies to editable workspaces
+		return "", fmt.Errorf("no environment name or ID provided and no default environment set, " +
+			"either pass in an environment name or ID or set a default environment by using `rad env switch`")
+	} else if environmentNameOrID == "" {
+		return "", fmt.Errorf("no environment name or ID provided, pass in an environment name or ID")
+	}
+
+	return environmentNameOrID, err
+}
+
 // DidSpecifyEnvironmentName checks if an environment name is provided as a flag
 func DidSpecifyEnvironmentName(cmd *cobra.Command, args []string) bool {
 	environmentName, err := cmd.Flags().GetString("environment")
-	return err == nil && environmentName != ""
+	return err == nil && environmentName != "" && !strings.HasPrefix(environmentName, resources.SegmentSeparator)
```

Comment:
Does this apply to both just the environment name and also the resource id for the environment?

Reply:
No, it checks only for environment name. Since now we might be dealing with an id, it is updated to make sure we dont have the SegmentSeperator character that suggests this is an id.

Reply:
Do we need `DidSpecifyEnvironmentID`?

Reply:
No, there wasn't necessity for it.